code/__HELPERS/mobs.dm

```diff
@@ -254,16 +261,23 @@ GLOBAL_LIST_EMPTY(species_list)
  * * target - the (optional) target mob of the do_after. If they move/cease to exist, the do_after is cancelled.
  * * timed_action_flags - optional flags to override certain do_after checks (see DEFINES/timed_action.dm).
  * * progress - if TRUE, a progress bar is displayed.
+ * * show_to_target - if TRUE, shows the bar and item to the target
  * * extra_checks - a callback that can be used to add extra checks to the do_after. Returning false in this callback will cancel the do_after.
+ * * add_item - shows said item being 'filled up' instead of the progress bar
+ * * x_offset - moves left/right increase/decrease for the item rendered above the bar
+ * * y_offset - moves up/down the item rendered above the bar
  */
-/proc/do_after(mob/user, delay = 3 SECONDS, atom/target, timed_action_flags = NONE, progress = TRUE, datum/callback/extra_checks)
+/proc/do_after(mob/user, delay = 3 SECONDS, atom/target, timed_action_flags = NONE, progress = TRUE, show_to_target = FALSE, \
```

Comment:
Does this progrss bar cancel if add_item gets deleted? I'm pretty sure appearance will count as a reference but im not entirely sure. You'll have to check DM docs